storagenode/trust/service.go

```diff
@@ -127,18 +127,26 @@ func (pool *Pool) GetSignee(ctx context.Context, id storj.NodeID) (_ signing.Sig
 		}
 	}
 
-	info.mu.Lock()
-	defer info.mu.Unlock()
+	info.mu.RLock()
+	identity := info.identity
+	info.mu.RUnlock()
 
-	if info.identity == nil {
-		identity, err := pool.kademlia.FetchPeerIdentity(ctx, id)
-		if err != nil {
-			return nil, Error.Wrap(err)
+	if identity == nil {
+		info.mu.Lock()
+		defer info.mu.Unlock()
+
+		// Check to see if another goroutine has won
+		identity = info.identity
```

Comment:
Do we need to check here again?

Reply:
Yes, in the case that there are two concurrent misses, only one wins the mutex, so the other should just happily use the value that the other stored.